From 06598c404e89bff27a0d57abb0def96c9dd0d1e7 Mon Sep 17 00:00:00 2001
From: Jerry Cao <jerry.cao@amlogic.com>
Date: Fri, 14 Mar 2014 15:28:55 -0700
Subject: [PATCH 3765/5965] [TrustZone] Update TEE kernel config for M8.

Also cleaned up M8 default kernel config (output from make defconfig).
---
 arch/arm/configs/meson8_defconfig     | 25 +++++++------------
 arch/arm/configs/meson8_tee_defconfig | 35 +++++++++++++++------------
 2 files changed, 28 insertions(+), 32 deletions(-)
 mode change 100755 => 100644 arch/arm/configs/meson8_tee_defconfig

diff --git a/arch/arm/configs/meson8_defconfig b/arch/arm/configs/meson8_defconfig
index 3fb0ccd927b0..5007095342b7 100755
--- a/arch/arm/configs/meson8_defconfig
+++ b/arch/arm/configs/meson8_defconfig
@@ -5,7 +5,6 @@ CONFIG_NO_HZ=y
 CONFIG_HIGH_RES_TIMERS=y
 CONFIG_BSD_PROCESS_ACCT=y
 CONFIG_IKCONFIG=y
-CONFIG_IKCONFIG_PROC=n
 CONFIG_LOG_BUF_SHIFT=19
 CONFIG_BLK_DEV_INITRD=y
 CONFIG_RD_BZIP2=y
@@ -26,11 +25,11 @@ CONFIG_PREEMPT=y
 CONFIG_AEABI=y
 CONFIG_HIGHMEM=y
 CONFIG_KSM=y
-CONFIG_AMLOGIC_MESON_CPUFREQ=y
 CONFIG_CPU_FREQ=y
 CONFIG_CPU_FREQ_GOV_ONDEMAND=y
 CONFIG_CPU_FREQ_GOV_INTERACTIVE=y
 CONFIG_CPU_FREQ_GOV_CONSERVATIVE=y
+CONFIG_AMLOGIC_MESON_CPUFREQ=y
 CONFIG_VFP=y
 CONFIG_NEON=y
 CONFIG_BINFMT_MISC=y
@@ -190,19 +189,19 @@ CONFIG_LIGHT_ISL29023=y
 CONFIG_LIGHT_STK220X=y
 CONFIG_LIGHT_LTR558=y
 CONFIG_AML_GPIO_KEY=y
-CONFIG_RICOH_PMU=y
+CONFIG_AML_PMU_ALGORITHM_SUPPORT=m
 CONFIG_AML_PMU=y
 CONFIG_AML1216=y
+CONFIG_RICOH_PMU=y
 CONFIG_SECURITYKEY=y
 CONFIG_EFUSE=y
 CONFIG_AMLOGIC_THERMAL=y
 CONFIG_AMLOGIC_USB=y
 CONFIG_MMC_AML=y
 CONFIG_AMLOGIC_SPI_NOR=y
-CONFIG_AML_NAND_KEY=y
 CONFIG_AML_NEXT_GEN_NAND=y
-CONFIG_AM_NAND=n
-CONFIG_AML_NFTL_NEW=m
+CONFIG_AML_NAND_KEY=y
+# CONFIG_AM_NAND is not set
 CONFIG_AM_ETHERNET=y
 CONFIG_AM_WIFI=y
 CONFIG_AM_WIFI_SD_MMC=y
@@ -226,8 +225,6 @@ CONFIG_FB_OSD2_DEFAULT_HEIGHT_VIRTUAL=32
 CONFIG_AM_FB_EXT=y
 CONFIG_AM_LOGO=y
 CONFIG_AM_HDMI_ONLY=y
-CONFIG_TVIN=y
-CONFIG_TVIN_VDIN=y
 CONFIG_VIUIN=y
 CONFIG_TVIN_ISP=y
 CONFIG_AML_SII9233=y
@@ -241,10 +238,10 @@ CONFIG_VIDEO_AMLOGIC_CAPTURE_GC0308=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_GC0328=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_GC2035=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_OV5640=y
-CONFIG_VIDEO_AMLOGIC_CAPTURE_SP0A19=y
-CONFIG_VIDEO_AMLOGIC_CAPTURE_SP1628=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_SP0838=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_SP2518=y
+CONFIG_VIDEO_AMLOGIC_CAPTURE_SP0A19=y
+CONFIG_VIDEO_AMLOGIC_CAPTURE_SP1628=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_HI253=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_HM2057=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_HM5065=y
@@ -258,6 +255,8 @@ CONFIG_AMLOGIC_IONVIDEO=y
 CONFIG_DEINTERLACE=y
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_CMA=y
+CONFIG_CMA_SIZE_MBYTES=8
 CONFIG_PROC_DEVICETREE=y
 CONFIG_OF_SELFTEST=y
 CONFIG_BLK_DEV_LOOP=y
@@ -354,9 +353,3 @@ CONFIG_SECURITY_NETWORK=y
 CONFIG_SECURITY_SELINUX=y
 CONFIG_CRC_T10DIF=y
 CONFIG_CRC7=y
-CONFIG_CMA=y
-CONFIG_CMA_SIZE_MBYTES=8
-CONFIG_CMA_SIZE_SEL_MBYTES=y
-CONFIG_CMA_ALIGNMENT=8
-CONFIG_AML_PMU_ALGORITHM_SUPPORT=m
-CONFIG_CMA_AREAS=7
diff --git a/arch/arm/configs/meson8_tee_defconfig b/arch/arm/configs/meson8_tee_defconfig
old mode 100755
new mode 100644
index 86c3323f4ed2..ef05b9147d3d
--- a/arch/arm/configs/meson8_tee_defconfig
+++ b/arch/arm/configs/meson8_tee_defconfig
@@ -1,9 +1,11 @@
 CONFIG_KERNEL_LZO=y
 CONFIG_SYSVIPC=y
+CONFIG_AUDIT=y
 CONFIG_NO_HZ=y
 CONFIG_HIGH_RES_TIMERS=y
 CONFIG_BSD_PROCESS_ACCT=y
 CONFIG_IKCONFIG=y
+CONFIG_LOG_BUF_SHIFT=19
 CONFIG_BLK_DEV_INITRD=y
 CONFIG_RD_BZIP2=y
 CONFIG_RD_LZMA=y
@@ -22,9 +24,7 @@ CONFIG_SMP=y
 CONFIG_PREEMPT=y
 CONFIG_AEABI=y
 CONFIG_HIGHMEM=y
-CONFIG_COMPACTION=y
 CONFIG_KSM=y
-#CONFIG_DEBUG_KMEMLEAK=y
 CONFIG_CPU_FREQ=y
 CONFIG_CPU_FREQ_GOV_ONDEMAND=y
 CONFIG_CPU_FREQ_GOV_INTERACTIVE=y
@@ -162,8 +162,7 @@ CONFIG_ADC_KEYPADS_AM=y
 CONFIG_MESON_INPUT_TOUCHSCREEN=y
 CONFIG_FOCALTECH_CAPACITIVE_TOUCHSCREEN=y
 CONFIG_GOODIX_GT81XX_CAPACITIVE_TOUCHSCREEN=y
-CONFIG_GSLX680_CAPACITIVE_TOUCHSCREEN=y
-CONFIG_GSLX680B_CAPACITIVE_TOUCHSCREEN=y
+CONFIG_GSLX680_COMPATIBLE_CAPACITIVE_TOUCHSCREEN=y
 CONFIG_GOODIX_GT9XX_CAPACITIVE_TOUCHSCREEN=y
 CONFIG_NOVATEK_CAPACITIVE_TOUCHSCREEN=y
 CONFIG_TOUCHSCREEN_CT36X=y
@@ -190,9 +189,10 @@ CONFIG_LIGHT_ISL29023=y
 CONFIG_LIGHT_STK220X=y
 CONFIG_LIGHT_LTR558=y
 CONFIG_AML_GPIO_KEY=y
-CONFIG_RICOH_PMU=y
+CONFIG_AML_PMU_ALGORITHM_SUPPORT=m
 CONFIG_AML_PMU=y
 CONFIG_AML1216=y
+CONFIG_RICOH_PMU=y
 CONFIG_SECURITYKEY=y
 CONFIG_EFUSE=y
 CONFIG_AMLOGIC_THERMAL=y
@@ -201,6 +201,7 @@ CONFIG_MMC_AML=y
 CONFIG_AMLOGIC_SPI_NOR=y
 CONFIG_AML_NEXT_GEN_NAND=y
 CONFIG_AML_NAND_KEY=y
+# CONFIG_AM_NAND is not set
 CONFIG_AM_ETHERNET=y
 CONFIG_AM_WIFI=y
 CONFIG_AM_WIFI_SD_MMC=y
@@ -213,6 +214,7 @@ CONFIG_BT_DEVICE=y
 CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_ENCODER=y
+CONFIG_AM_JPEG_ENCODER=y
 CONFIG_AM_TV_OUTPUT2=y
 CONFIG_FB_AM=y
 CONFIG_FB_OSD2_CURSOR=y
@@ -221,11 +223,11 @@ CONFIG_FB_OSD2_DEFAULT_HEIGHT=32
 CONFIG_FB_OSD2_DEFAULT_WIDTH_VIRTUAL=32
 CONFIG_FB_OSD2_DEFAULT_HEIGHT_VIRTUAL=32
 CONFIG_AM_FB_EXT=y
-CONFIG_AMLOGIC_BACKLIGHT=y
 CONFIG_AM_LOGO=y
 CONFIG_AM_HDMI_ONLY=y
 CONFIG_VIUIN=y
 CONFIG_TVIN_ISP=y
+CONFIG_AML_SII9233=y
 CONFIG_POST_PROCESS_MANAGER=y
 CONFIG_POST_PROCESS_MANAGER_PPSCALER=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE=y
@@ -236,16 +238,17 @@ CONFIG_VIDEO_AMLOGIC_CAPTURE_GC0308=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_GC0328=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_GC2035=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_OV5640=y
-CONFIG_VIDEO_AMLOGIC_CAPTURE_SP0A19=y
-CONFIG_VIDEO_AMLOGIC_CAPTURE_SP1628=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_SP0838=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_SP2518=y
+CONFIG_VIDEO_AMLOGIC_CAPTURE_SP0A19=y
+CONFIG_VIDEO_AMLOGIC_CAPTURE_SP1628=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_HI253=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_HM2057=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_HM5065=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_OV5647=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_AR0543=y
 CONFIG_VIDEO_AMLOGIC_CAPTURE_AR0833=y
+CONFIG_VIDEO_AMLOGIC_CAPTURE_BF3720=y
 CONFIG_V4L_AMLOGIC_VIDEO=y
 CONFIG_V4L_AMLOGIC_VIDEO2=y
 CONFIG_AMLOGIC_IONVIDEO=y
@@ -253,6 +256,8 @@ CONFIG_DEINTERLACE=y
 CONFIG_MESON_TRUSTZONE=y
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_CMA=y
+CONFIG_CMA_SIZE_MBYTES=8
 CONFIG_PROC_DEVICETREE=y
 CONFIG_OF_SELFTEST=y
 CONFIG_BLK_DEV_LOOP=y
@@ -262,7 +267,6 @@ CONFIG_BLK_DEV_SR=y
 CONFIG_SCSI_MULTI_LUN=y
 CONFIG_NETDEVICES=y
 CONFIG_TUN=y
-CONFIG_PHYLIB=y
 CONFIG_SMSC_PHY=y
 CONFIG_ICPLUS_PHY=y
 CONFIG_PPP=y
@@ -276,6 +280,7 @@ CONFIG_PPPOPNS=y
 CONFIG_PPP_ASYNC=y
 CONFIG_PPP_SYNC_TTY=y
 CONFIG_USB_USBNET=y
+CONFIG_USB_NET_DM9601=y
 CONFIG_INPUT_POLLDEV=y
 CONFIG_INPUT_EVDEV=y
 CONFIG_INPUT_MISC=y
@@ -289,8 +294,6 @@ CONFIG_BACKLIGHT_LCD_SUPPORT=y
 CONFIG_SND_USB_AUDIO=y
 CONFIG_SND_AML_M8_SOC=y
 CONFIG_SND_AML_M8=y
-CONFIG_SND_AML_M8_PCM=y
-CONFIG_SND_AML_M8_RT5616=y
 CONFIG_UHID=y
 CONFIG_USB=y
 CONFIG_USB_ACM=y
@@ -305,11 +308,10 @@ CONFIG_UIO=y
 CONFIG_UIO_PDRV=y
 CONFIG_UIO_PDRV_GENIRQ=y
 CONFIG_STAGING=y
-CONFIG_ANDROID=y
 CONFIG_ZSMALLOC=y
 CONFIG_ZRAM=y
 CONFIG_ZRAM_DEBUG=y
-CONFIG_KSM=y
+CONFIG_ANDROID=y
 CONFIG_ANDROID_BINDER_IPC=y
 CONFIG_ASHMEM=y
 CONFIG_ANDROID_LOGGER=y
@@ -342,12 +344,13 @@ CONFIG_NLS_UTF8=y
 CONFIG_PRINTK_TIME=y
 CONFIG_MAGIC_SYSRQ=y
 CONFIG_READABLE_ASM=y
+CONFIG_DEBUG_FS=y
 CONFIG_DEBUG_SECTION_MISMATCH=y
 CONFIG_SCHEDSTATS=y
 CONFIG_TIMER_STATS=y
 CONFIG_DEBUG_INFO=y
+CONFIG_SECURITY=y
+CONFIG_SECURITY_NETWORK=y
+CONFIG_SECURITY_SELINUX=y
 CONFIG_CRC_T10DIF=y
 CONFIG_CRC7=y
-CONFIG_EMMC_SECURE_STORAGE=y
-CONFIG_DEBUG_FS=y
-CONFIG_AMLOGIC_THERMAL=y
-- 
2.19.0

